tests: logging: log_backend_uart: fix build error with MWDT toolchain

**Describe the bug**

Currently we get the build error when building `tests/subsys/logging/log_backend_uart` with MWDT toolchain:

```
/verification_workspace/zephyr/tests/subsys/logging/log_backend_uart/src/main.c:22:14: error: static_assert expression is not an integral constant expression
BUILD_ASSERT(strlen(TEST_DATA) < SAMPLE_DATA_SIZE);
~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/verification_workspace/zephyr/include/zephyr/toolchain/mwdt.h:151:52: note: expanded from macro 'BUILD_ASSERT'
#define BUILD_ASSERT(EXPR, MSG...) _Static_assert((EXPR), "" MSG)
                                                  ~^~~~~
```

**Expected behavior**
No build error should happen

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: MWDT 2024.06
 - Commit SHA: recent main, 1ed04e7b49c5bec8e09f81480b7f95ae51b59b10

--- a/tests/subsys/logging/log_backend_uart/src/main.c
+++ b/tests/subsys/logging/log_backend_uart/src/main.c
@@ -19,7 +19,8 @@ LOG_MODULE_REGISTER(test, CONFIG_SAMPLE_MODULE_LOG_LEVEL);
 #define SAMPLE_DATA_SIZE          EMUL_UART_TX_FIFO_SIZE(0)
 
 #define TEST_DATA "0123456789ABCDEF"
-BUILD_ASSERT(strlen(TEST_DATA) < SAMPLE_DATA_SIZE);
+/* Assert that size of the test string (without '\0') fits in the SAMPLE_DATA_SIZE */
+BUILD_ASSERT((sizeof(TEST_DATA) - 1) < SAMPLE_DATA_SIZE);
 
 struct log_backend_uart_fixture {
 	const struct device *dev[EMUL_UART_NUM];

